ui/app/mixins/sortable-factory.js

```diff
@@ -0,0 +1,42 @@
+import Mixin from '@ember/object/mixin';
+import { computed } from '@ember/object';
+
+/**
+  Sortable mixin factory
+
+  Simple sorting behavior for a list of objects. Pass the list of properties
+  you want the list to be live-sorted based on, or use the generic sortable.js
+  if you don’t need that.
+
+  Properties to override:
+    - sortProperty: the property to sort by
+    - sortDescending: when true, the list is reversed
+    - listToSort: the list of objects to sort
+
+  Properties provided:
+    - listSorted: a copy of listToSort that has been sorted
+*/
+export default function sortableFactory(properties) {
+  const eachProperties = properties.map(property => `listToSort.@each.${property}`);
```

Comment:
I agree with your concern that this problem should be fixed for all tables, and I also agree your resistance to dig into all of them at once.

Maybe a good middle ground just so the problem isn't forgotten about once this merges is to `warn` here if `properties.length === 0`. The warning could also include something like "The Sortable mixin is deprecated in favor of SortableFactory".

Reply:
👍 thanks for the suggestion; I added something in 0131c36, it’s somewhat unwieldy because of when mixins are mixed in, but it works, thoughts?

Reply:
Looks like it'll do, but I agree it's unwieldy. What's preventing you from putting it right here, in the `sortableFactory` but not in the returned `Mixin`? Don't we have all the information at this point? It would also prevent the state tracking needed right now.

Reply:
oops, I merged this 😳 I’m sorry

I think that would have the same problem though because the function is called when the mixin is created, which happens when the application loads, so the warnings all happen at the same time.